smack-experimental/src/main/java/org/jivesoftware/smackx/push_notifications/PushNotificationsManager.java

```diff
@@ -0,0 +1,181 @@
+                .serverSupportsFeature(PushNotificationsElements.NAMESPACE);
+    }
+
+    /**
+     * Enable push notifications.
+     * 
+     * @param pushJid
+     * @param node
+     * @return true if it was successfully enabled, false if not
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public boolean enable(Jid pushJid, String node)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        return enable(pushJid, node, null);
+    }
+
+    /**
+     * Enable push notifications.
+     * 
+     * @param pushJid
+     * @param node
+     * @param publishOptions
+     * @return true if it was successfully enabled, false if not
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public boolean enable(Jid pushJid, String node, HashMap<String, String> publishOptions)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        EnablePushNotificationsIQ enablePushNotificationsIQ = new EnablePushNotificationsIQ(pushJid, node,
+                publishOptions);
+        return changePushNotificationsStatus(enablePushNotificationsIQ);
+    }
+
+    /**
+     * Disable all push notifications.
+     * 
+     * @param pushJid
+     * @return true if it was successfully disabled, false if not
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public boolean disableAll(Jid pushJid)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        return disable(pushJid, null);
+    }
+
+    /**
+     * Disable push notifications of an specific node.
+     * 
+     * @param pushJid
+     * @param node
+     * @return true if it was successfully disabled, false if not
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public boolean disable(Jid pushJid, String node)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        DisablePushNotificationsIQ disablePushNotificationsIQ = new DisablePushNotificationsIQ(pushJid, node);
+        return changePushNotificationsStatus(disablePushNotificationsIQ);
+    }
+
+    private boolean changePushNotificationsStatus(IQ iq)
+            throws NotConnectedException, InterruptedException, NoResponseException, XMPPErrorException {
+        final XMPPConnection connection = connection();
+
+        IQ responseIQ = null;
+        PacketCollector responseIQCollector = connection.createPacketCollector(new IQReplyFilter(iq, connection));
```

Comment:
For IQ request/response sequences simply use

```java
connection.createPacketCollectorAndSend(iq).nextResultOrThrow();
```

this combination takes care of canceling the collector in any case. Reduces the LOC by ~10 here.